enumerable/socket.go

```diff
@@ -0,0 +1,62 @@
+package enumerable
+
+import "github.com/sourcenetwork/immutable"
+
+// Socket is an extention of the enumerable interface allowing the source
+// to be replaced after initial construction.
+type Socket[T any] interface {
```

Comment:
question: Can you describe the usecase for this particular type?

Reply:
You can see this being used here: https://github.com/sourcenetwork/defradb/pull/1564/files#diff-e7f1317923231a3017c3a580d21ce39346555446638d045a69310b1a1aeab15d in the MigrateUp function.  It allows  enumerables to be defined before their source.